libc/src/__support/invasive_list.h

```diff
@@ -0,0 +1,60 @@
+//===-- Invasive queue implementation. --------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// An invasive list that implements the insque and remque semantics.
+//
+//===----------------------------------------------------------------------===//
+
+
+#ifndef LLVM_LIBC_SRC___SUPPORT_INVASIVE_QUEUE_H
+#define LLVM_LIBC_SRC___SUPPORT_INVASIVE_QUEUE_H
+
+#include "common.h"
+
+namespace LIBC_NAMESPACE {
+namespace internal {
+
+struct InvasiveList {
+  struct InvasiveNodeHeader {
```

Comment:
What do you mean by `Invasive`? I've heard of `Intrusive` data structures, but not `Invasive` data structures.

Reply:
I suppose `Intrusive` is more standard usage.

Reply:
Renamed to `IntrusiveList`.